src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -813,72 +811,72 @@ public string GetDisplayName(SyntaxNode? node, DisplayNameOptions options, strin
-                        builder.Append(name);
-                        AppendTypeParameterList(builder, memberDeclaration.GetTypeParameterList());
-                        name = pooled.ToStringAndFree();
-                    }
-                }
-                else
-                {
-                    Debug.Assert(memberDeclaration.Kind() == SyntaxKind.IncompleteMember);
-                    name = "?";
-                }
+                if ((options & DisplayNameOptions.IncludeTypeParameters) != 0)
+                    AppendTypeParameterList(builder, memberDeclaration.GetTypeParameterList());
+
+                AppendParameterList(builder, extensionDeclaration.ParameterList);
+                return builder.ToString();
             }
-        }
-        else
-        {
-            if (node is VariableDeclaratorSyntax fieldDeclarator)
+            else
             {
-                var nameToken = fieldDeclarator.Identifier;
-                if (nameToken != default)
-                {
-                    name = nameToken.IsMissing ? missingTokenPlaceholder : nameToken.Text;
-                }
+                Debug.Assert(memberDeclaration.Kind() == SyntaxKind.IncompleteMember);
+                return "?";
             }
         }
+        else if (node is VariableDeclaratorSyntax fieldDeclarator)
+        {
+            var nameToken = fieldDeclarator.Identifier;
+            return nameToken.IsMissing ? missingTokenPlaceholder : nameToken.Text;
+        }
 
-        Debug.Assert(name != null, "Unexpected node type " + node.Kind());
-        return name;
-    }
+        Debug.Fail("Unexpected node type " + node.Kind());
+        return null;
 
-    private static void AppendTypeParameterList(StringBuilder builder, TypeParameterListSyntax typeParameterList)
-    {
-        if (typeParameterList != null && typeParameterList.Parameters.Count > 0)
+        static void AppendTypeParameterList(StringBuilder builder, TypeParameterListSyntax? typeParameterList)
         {
-            builder.Append('<');
-            builder.Append(typeParameterList.Parameters[0].Identifier.ValueText);
-            for (var i = 1; i < typeParameterList.Parameters.Count; i++)
+            if (typeParameterList != null)
             {
-                builder.Append(", ");
-                builder.Append(typeParameterList.Parameters[i].Identifier.ValueText);
+                builder.Append('<');
+                builder.Append(string.Join(", ", typeParameterList.Parameters.Select(static p => p.Identifier.ValueText)));
+                builder.Append('>');
             }
+        }
 
-            builder.Append('>');
+        static void AppendParameterList(StringBuilder builder, ParameterListSyntax? parameterList)
+        {
+            if (parameterList != null)
+            {
+                builder.Append('(');
+                builder.Append(string.Join(", ", parameterList.Parameters.Select(static p => Instance.ConvertToSingleLine(p.Type))));
```

Comment:
>Instance [](http://example.com/codeflow?start=93&length=8)

another usage of Instance.ConvertToSingleLine that could be simplified if the method weren't static